Primo.js

```diff
@@ -278,32 +287,87 @@ function importPNX(text) {
+		// Should also catch cases like "Meurs, Jacob : van "
+		item.publisher = publisher;
+	} else if(pubplace && pubplace[1]) {
+		// Hopefully we split in place, publisher name.
 		item.publisher = pubplace[1].replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "")
 			.replace(/^\s*"|,?"\s*$/g, '');
 	} else if(pubplace) {
-		item.publisher = pubplace[0].replace(/,\s*c?\d+|[\(\)\[\]]|(\.\s*)?/g, "")
-			.replace(/^\s*"|,?"\s*$/g, '');
+		// No splitting, use the whole (cleaned) string
+		item.publisher = possibleplace.replace(/^\s*"|,?"\s*$/g, '');
 	}
-	
+	if(!item.place && possibleplace && pubplace[1]) {
+		// Splitting happened and we need a place, we hope this is it
+		item.place = possibleplace;
+	}
+
+	// //addata/date is sometimes available: ever needed?
+	// //addata/risdate can contain the date as written on the manifestation and
+	// be the only date, e.g. "Anno domini MCCCCLXXXXV die XXVII februarii" in
+	// http://digitale.beic.it/primo_library/libweb/action/display.do?doc=39bei_digitool2394625&showPnx=true
 	var date = ZU.xpathText(doc, '//display/creationdate|//search/creationdate');
 	var m;
 	if(date && (m = date.match(/\d+/))) {
 		item.date = m[0];
 	}
-	
-	// the three letter ISO codes that should be in the language field work well:
+
+	// Old publishers may be written as "Last, First" etc. according to
+	// standards, but should be like authors when in citations.
+	// Source: Italian librarian https://it.wikipedia.org/?diff=69781295
+	// "Dozza, Evangelista (1.), eredi"
+	// "Marnef, Jérôme de & Cavellat, Guillaume, veuve"
+	// But:
+	// "The Minerals, Metals & Materials Society"
+	// "Hodges, Foster & Co."
+	// Also problematic and still to handle: annotations which are not proper names, like:
+	// "printed by Henry Hills, printer to the King's most Excellent Majesty, for His Household and Chappel"
+	// "Hessisches Ministerium für Landwirtschaft, Forsten und Naturschutz, Landentwicklung"
+	if( item.publisher
+		&& item.date
+		&& item.date < '1831'
+	) {
+		// Remove excess space and secondary disambiguations
+		// Do not replace "&" unless there is a second name-looking string
+		var declutter = ZU.trimInternal(item.publisher).replace(/ \& ([^&,;]+,)/g, "; $1").replace(/ *\([^)]+\)/g, "");
+		// Shuffle names with disambiguation and remove numbering
+		var shuffle3 = declutter.replace(/ *([^,;]+), +([^,;0-9]+) *[0-9]*, +([^,;]+) */g, "$2 $1 ($3)");
+		// Same, other names
+		var shuffle2 = shuffle3.replace(/ *([^,;]+), +([^,;0-9]+) *[0-9]* */g, "$2 $1");
+		// Use comma list, get rid of space buildup for numbered names, clean like authors
+		item.publisher = stripAuthor( shuffle2.replace(/ *[;] */g, ", ").replace(/  /g, " ") );
+	}
+
+	// The three letter codes, that should be in the language field, usually work well
+	// (although they may be in MARC21, or ISO 639-2, rather than ISO 639-3 or others)
 	item.language = ZU.xpathText(doc, '(//display/language|//facets/language)[1]');
```

Comment:
Philipp Zumstein, 04/05/2015 21:41:
> I don't understand the two comment lines above here.

The language codes are not in ISO 639-3. That's subject of my other open 
PR, https://github.com/zotero/translators/pull/835